src/dali/plugin/pair_converter/ccxt.py

```diff
@@ -295,6 +270,58 @@ def get_historic_bar_from_native_source(self, timestamp: datetime, from_asset: s
 
         return result
 
+    def _cache_exchange_data(self) -> None:
+        if self.__exchanges:
+            self.__logger.debug("Exchange data already cached.")
+            return None
+
+        # first, initialize all exchanges so we use a consistent initialization approach
+        for exchange in _EXCHANGE_DICT:
+            # initializes the cctx exchange instance which is used to get the historical data
+            # https://docs.ccxt.com/en/latest/manual.html#notes-on-rate-limiter
+            self.__exchanges[exchange] = _EXCHANGE_DICT[exchange]({"enableRateLimit": True})
```

Comment:
`enableRateLimit` is enabled by default. Have you had more success with it explicitly enabled? It didn't affect anything for me when I tested it. I still had issues with Kraken blocking me.

Reply:
Ah, ok, didn't realize that. I was having some weird ddos/timeout issues, and they went away after enabling this, but it could have been because I waited a day to retry.